translatable_proc/src/data/translations.rs

```diff
@@ -1,57 +1,60 @@
-use crate::languages::Iso639a;
-use crate::translations::errors::TranslationError;
+use super::config::{ConfigError, SeekMode, TranslationOverlap, load_config};
 
-/// Errors occurring during TOML-to-translation structure transformation
+/// Contains error definitions for what may go wrong while
+/// loading a translation.
 #[derive(Error, Debug)]
-pub enum TransformError {
-    /// Mixed content found in nesting node (strings and objects cannot coexist)
-    #[error("A nesting can contain either strings or other nestings, but not both.")]
-    InvalidNesting,
-
-    /// Template syntax error with unbalanced braces
-    #[error("Templates in translations should match '{{' and '}}'")]
-    UnclosedTemplate,
-
-    /// Invalid value type encountered in translation structure
-    #[error("Only strings and objects are allowed for nested objects.")]
-    InvalidValue,
-
-    /// Failed to parse language code from translation key
-    #[error("Couldn't parse ISO 639-1 string for translation key")]
-    LanguageParsing(#[from] ParseError),
-}
-
-/// Represents hierarchical translation structure
-#[derive(Clone)]
-pub enum NestingType {
-    /// Nested namespace containing other translation objects
-    Object(HashMap<String, NestingType>),
-    /// Leaf node containing actual translations per language
-    Translation(HashMap<Iso639a, String>),
-}
-
-/// Translation association with its source file
-pub struct AssociatedTranslation {
-    /// Original file path of the translation
-    original_path: String,
-    /// Hierarchical translation data
-    translation_table: NestingType,
+pub enum TranslationDataError {
+    /// Represents a generic IO error, if a file couldn't
+    /// be opened, a path could not be found... This error
+    /// will be inferred from an `std::io::Error`.
+    #[error("There was a problem with an IO operation: {0:#}")]
+    SystemIo(#[from] IoError),
+
+    /// Used to convert any configuration loading error
+    /// into a `TranslationDataError`, error messages are
+    /// handled by the `ConfigError` itself.
+    #[error("{0:#}")]
+    LoadConfig(#[from] ConfigError),
+
+    /// Specific conversion error for when a path can't be converted
+    /// to a manipulable string because it contains invalid
+    /// unicode characters.
+    #[error("Couldn't open path, found invalid unicode characters")]
+    InvalidUnicode,
+
+    /// Represents a TOML deserialization error, this happens while
+    /// loading files and converting their content to TOML.
+    ///
+    /// # Arguments
+    /// * `.0` - The RAW toml::de::Error returned by the deserialization
+    /// function.
+    /// * `.1` - The path where the file was originally found.
+    #[error(
+        "TOML Deserialization error '{reason}' {span} in {1}",
+        reason = _0.message(),
+        span = _0
+            .span()
+       .map(|range| format!("on {}:{}", range.start, range.end))
+            .unwrap_or_else(|| String::new())
```

Comment:
Remove redundant closure.

```suggestion
            .unwrap_or_else(String::new)
```